cmds/cmp/cmp.go

```diff
@@ -76,34 +91,29 @@ func main() {
 
 	fnames := flag.Args()
 
-	switch len(fnames) {
-	case 2:
-	case 3:
-		offset[0], err = strconv.ParseInt(fnames[2], 0, 64)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "bad offset1: %s: %v\n", fnames[2], err)
-			return
-		}
-	case 4:
-		offset[0], err = strconv.ParseInt(fnames[2], 0, 64)
-		if err != nil {
-			log.Printf("bad offset1: %s: %v\n", fnames[2], err)
-			return
-		}
-		offset[1], err = strconv.ParseInt(fnames[3], 0, 64)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "bad offset2: %s: %v\n", fnames[3], err)
-			return
-		}
-	default:
-		log.Fatalf("expected two filenames (and one to two optional offsets), got %d", len(fnames))
+	if nArgs := len(fnames); nArgs < 2 {
+		failf("Too few arguments, expected at least two filenames, got %d arguments", nArgs)
+	} else if nArgs > 4 {
+		failf("Too many arguments, expected exactly two filenames (and one or two optional offsets), got %d arguments", nArgs)
+	}
+
+	cmpUnits := unit.DefaultUnits
+	cmpUnits["kB"] = unit.DefaultUnits["KB"]
+
+	off, err := unit.NewUnit(cmpUnits)
+	if err != nil {
+		failf("Could not create unit based on mapping: %v\n", err)
+	}
+
+	for i := len(fnames) - 2; i > 0; i-- {
+		setOffsetOrFail(fnames[i+1], off, &offset[i-1])
```

Comment:
this seems much less intuitive than the switch statement we had before. I'd rather not have to do any mental math with offsets; esp when we are limited to 2-4 elements max anyway.

Reply:
Okay, here we disagree. The old code had:
- an empty case
- code duplication
- collected all the possible errors in the default case. Probably I changed/split it because of that, because I remembered that Ron does not like these "guess what really went wrong" semantic.

I find the offset calculation not too difficult, and it works nicely with 1 or 2 offsets.

But that is my opinion and as you make the rules, I will re-introduce the IMO ugly switch again. As I don't have time till Mon to work on it again, let's see if others have an opinion, otherwise I roll it back.

Reply:
The for loop is confusing to me and makes people have to go find that function. I think I'm ok with the switch, personally.